RadClass/RadClass.py

```diff
@@ -151,14 +160,24 @@ def march(self):
 
         return running
 
+    def run_cache(self):
+        start_i, = np.where(self.processor.timestamps == self.working_time)
```

Comment:
Should we be keeping `start_i` up-to-date as a state variable of `self` rather than having to re-find it periodically?

Reply:
`start_i` is already kept up-to-date as a state variable, in a way, as `self.working_time`, which is the value equivalent of the index, `start_i`. Similar operations happen in liones 116 and 146. I could collapse all of these operations to one by adding an `np.where()` execution in `march()` when `self.working_time` is updated?

Reply:
Or increment `start_i` by some amount each time you use it?